common/src/main/scala/org/apache/celeborn/common/util/Utils.scala

```diff
@@ -415,8 +415,21 @@ object Utils extends Logging {
     customHostname = Some(hostname)
   }
 
-  def localHostName(conf: CelebornConf): String = customHostname.getOrElse {
-    if (conf.bindPreferIP) {
+  def localHostName(conf: CelebornConf): String = {
+    getHostName(conf.bindPreferIP)
+  }
+
+  def localHostNameForAdvertiseAddress(conf: CelebornConf): String = {
+    if (conf.masterHost != "<localhost>") {
```

Comment:
We should separate the logic for obtaining the AdvertiseAddress for the master and worker nodes.

Reply:
fixed